shared/types/guards.ts

```diff
@@ -47,21 +49,59 @@ export const isReward = (a: any): a is Reward =>
   typeof a.amount === 'number';
 
 export const isActivation = (a: any): a is Activation =>
-  a && a.id && a.layer && a.smesherId && a.coinbase && a.numUnits;
+  Boolean(a && a.id && a.layer && a.smesherId && a.coinbase && a.numUnits);
 
 export const isNodeError = (a: any): a is NodeError =>
-  a && a.msg && a.module && a.level;
+  Boolean(a && a.msg && a.module && a.level);
 
 export const isWalletSecretsEncrypted = (a: any): a is WalletSecretsEncrypted =>
-  a && a.cipher && a.cipherText;
+  Boolean(a && a.cipher && a.cipherText);
 
 export const isWalletGCMEncrypted = (a: any): a is WalletSecretsEncryptedGCM =>
-  isWalletSecretsEncrypted(a) && a.cipher === 'AES-GCM';
+  isWalletSecretsEncrypted(a) &&
+  a.cipher === 'AES-GCM' &&
+  typeof a.cipherParams?.iv === 'string' &&
+  a.kdf === 'PBKDF2' &&
+  typeof a.kdfparams?.dklen === 'number' &&
+  a.kdfparams?.hash === 'SHA-512' &&
+  typeof a.kdfparams?.iterations === 'number' &&
+  typeof a.kdfparams?.salt === 'string';
 
 export const isWalletLegacyEncrypted = (
   a: any
 ): a is WalletSecretsEncryptedLegacy =>
   isWalletSecretsEncrypted(a) && a.cipher === 'AES-128-CTR';
 
 export const isWalletSecrets = (a: any): a is WalletSecrets =>
-  a && a.mnemonic && a.accounts && a.contacts;
+  Boolean(a && a.mnemonic && a.accounts && a.contacts);
+
+export const isWalletHasStructure = (a: any): a is WalletFile =>
+  Boolean(a && a.meta && a.crypto);
+
+export const isWalletMeta = (meta: WalletMeta) =>
+  typeof meta.displayName === 'string' &&
+  typeof meta.created === 'string' &&
+  typeof meta.remoteApi === 'string' &&
+  typeof meta.type === 'string';
+
+export const isWalletFile = (wallet: WalletFile) => {
+  try {
+    if (!isWalletHasStructure(wallet)) {
+      return false;
+    }
+    const { crypto, meta } = wallet;
+
+    const isWalletMetaAndSecretsExists =
+      isWalletMeta(meta) && isWalletSecretsEncrypted(crypto);
+
+    if (!isWalletMetaAndSecretsExists) {
+      return false;
+    }
+
+    return isWalletGCMEncrypted(crypto) || isWalletLegacyEncrypted(crypto);
+  } catch (error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+    return false;
+  }
+};
```

Comment:
1. All guards are pure functions with object property checks. No need to wrap it into try-catch, if there is some exception has occurred — it means that there is a bug in the code.
2. It is not a TS guard.

```suggestion
export const isWalletFile = (wallet: any): wallet is WalletFile => (
   wallet &&
   isWalletMeta(wallet.meta) &&
   isWalletSecretsEncrypted(wallet.crypto) &&
   (
     isWalletGCMEncrypted(wallet.crypto) ||
     isWalletLegacyEncrypted(wallet.crypto)
   );
```